bentoml/frameworks/transformers.py

```diff
@@ -0,0 +1,164 @@
+    >>>          input_ids = tokenizer.encode(src_text, return_tensors="pt")
+    >>>          output = model.generate(input_ids, max_length=50)
+    >>>          output = [tokenizer.decode(output[0], skip_special_tokens=True)]
+    >>>          return output
+    >>>
+    >>>
+    >>> svc = TransformersService()
+    >>> ts = TransformersGPT2TextGenerator()
+    >>> tokenizer = AutoTokenizer.from_pretrained("gpt2")
+    >>> model = AutoModelWithLMHead.from_pretrained("gpt2", pad_token_id=tokenizer.eos_token_id)
+    >>> ts.pack("gptModel", {"model": model, "tokenizer": tokenizer})
+    >>> saved_path = ts.save()
+    """
+
+    def __init__(self, name):
+        super(TransformersModelArtifact, self).__init__(name)
+        self._model = None
+        self._model_type = None
+        self._tokenizer_type = None
+
+    def _file_path(self, base_path):
+        return os.path.join(base_path, self.name)
+
+    def _load_from_directory(self, path):
+        if self._model_type is None:
+            raise NotFound(
+                "Type of transformers model not found. This should be present in a file called 'model_type.txt' in the artifacts of the bundle."
+            )
+        if self._tokenizer_type is None:
+            raise NotFound(
+                "Type of transformers tokenizer not found. This should be present in a file called 'tokenizer_type.txt' in the artifacts of the bundle."
+            )
+        transformers_model = eval(self._model_type).from_pretrained(path)
+        tokenizer = eval(self._tokenizer_type).from_pretrained(path)
+        self._model = {"model": transformers_model, "tokenizer": tokenizer}
+
+    def _load_from_dict(self, model):
+        if not model.get("model"):
+            raise InvalidArgument(
+                " 'model' key is not found in the dictionary. Expecting a dictionary of with keys 'model' and 'tokenizer'"
+            )
+        if not model.get("tokenizer"):
+            raise InvalidArgument(
+                "'tokenizer' key is not found in the dictionary. Expecting a dictionary of with keys 'model' and 'tokenizer'"
+            )
+
+        model_class = str(type(model.get("model")).__module__)
+        tokenizer_class = str(type(model.get("tokenizer")).__module__)
+        # if either model or tokenizer is not an object of transformers
+        if not model_class.startswith("transformers"):
+            raise InvalidArgument(
+                "Expecting a transformers model object but object passes is {}".format(
+                    type(model.get("model"))
+                )
+            )
+        if not tokenizer_class.startswith("transformers"):
+            raise InvalidArgument(
+                "Expecting a transformers model object but object passes is {}".format(
+                    type(model.get("tokenizer"))
+                )
+            )
+        # success
+        self._model = model
+
+    def _load_from_string(self, model_name):
```

Comment:
👍 this is very cool! would be great to add an example that uses a string to pack pretrained model

Reply:
I've added it in the documentation under the  'example usage' section for the artifact.